optimum/modeling_base.py

```diff
@@ -0,0 +1,218 @@
+import json
+import logging
+import os
+import subprocess
+from abc import ABC, abstractmethod
+from typing import Optional, Union
+
+from transformers import AutoConfig
+
+import requests
+from huggingface_hub import HfApi, HfFolder, hf_hub_download
+
+from .utils import CONFIG_NAME
+
+
+logger = logging.getLogger(__name__)
+
+
+class OptimizedModel(ABC):
+    config_class = AutoConfig
+    load_tf_weights = None
+    base_model_prefix = "optimized_model"
+
+    def __init__(self, model=None, config=None, **kwargs):
+        self.model = model
+        self.config = config
+
+    def __call__(self, *args, **kwargs):
+        return self.forward(*args, **kwargs)
+
+    @abstractmethod
+    def forward(self, *args, **kwargs):
+        """
+        Forward pass of the model, needs to be overwritten.
+        """
+        pass
+
+    def save_pretrained(
+        self,
+        save_directory: Union[str, os.PathLike],
+        push_to_hub: bool = False,
+        **kwargs,
+    ):
+        """
+        Save a model and its configuration file to a directory, so that it can be re-loaded using the
+        `:func:`~optimized_pipelines.NeuronModel.from_pretrained`` class method.
+        Arguments:
+            save_directory (:obj:`str` or :obj:`os.PathLike`):
```

Comment:
Similarly here, we can drop the Sphinx syntax - check out the `doc-builder` guide [here](https://github.com/huggingface/doc-builder#writing-documentation-for-hugging-face-libraries) for conventions etc: